src/net/chat.rs

```diff
@@ -1068,6 +1068,9 @@ impl ConversationP2P {
     /// Handle an inbound authenticated p2p data-plane message.
     /// Return the message if not handled
     fn handle_data_message(&mut self, local_peer: &LocalPeer, peerdb: &mut PeerDB, burndb: &BurnDB, chainstate: &mut StacksChainState, chain_view: &BurnchainView, msg: StacksMessage) -> Result<Option<StacksMessage>, net_error> {
+        #[cfg(feature = "monitoring")]
```

Comment:
**Please do not pepper the codebase with conditionally-compiled code snippets!**  It significantly degrades readability, and it makes it needlessly difficult to swap out one monitoring endpoint implementation from another.  Instead, please encapsulate these snippets in a separate module, and replace these snippets' call sites with a function call instead of a bare feature-enabled snippet.

Reply:
Addressed with https://github.com/blockstack/stacks-blockchain/pull/1670/commits/43317acdb83474ae49ef9f2b8d2b7aad21080607